### PR TITLE
raw to Engineering value calibration

### PR DESCRIPTION
**Pull request by nicHoch**
generic framework to calibrate parameters from raw to engineering values based on the IDB

additional calibration steps might be necessary for higher product levels

### CODE REVIEW

File: stixcore/idb/idb.py [hunk 1]
```diff
@@ -138,6 +140,83 @@ def is_variable(self):
         return self.PID_TPSD != -1
 
 
+class IDBPolynomialCalibration(IdbData):
+    def __init__(self, rows):
```
Comment (samaloney) [inline stixcore/idb/idb.py:144]: What is row a list of coefficients?

File: stixcore/idb/idb.py [hunk 2]
```diff
@@ -395,6 +474,65 @@ def is_variable(self):
...
+        return self.__dict__['PCF_CURTX']
+
+    @property
+    def PCF_CATEG(self):
+        """Calibration category of the parameter.
+
+        Returns
+        -------
+        `str`
+            N|S|T|R|D|P|H|S|C
```
Comment (samaloney) [inline stixcore/idb/idb.py:520]: Could be handy to have what these mean in  the docstring

File: stixcore/idb/manager.py [hunk 3]
```diff
@@ -225,6 +226,39 @@ def convert_mib_2_sqlite(*, in_folder, out_file, version_label):
...
+                nix = 'NIX00405'
+                count,  = cur.execute("select count(*) from PCF where PCF_NAME = ? " +
+                                      "AND PCF_CURTX not NULL", (nix,))\
+                             .fetchone()
+                if count == 0:
+                    pname, nextID = IDB.generate_calibration_name("CIX", nextID)
+                    cur.execute('''update PCF set
+                                    PCF_CURTX = ?,
+                                    PCF_CATEG = 'N',
+                                    PCF_UNIT = 's'
```
Comment (samaloney) [inline stixcore/idb/manager.py:250]: is there a format here like `VOLT` should this be `sec` or just `s`?

File: stixcore/idb/tests/test_idb.py [hunk 4]
```diff
@@ -230,18 +193,19 @@ def test_textual_interpret(idb):
...
     info = idb.textual_interpret('foobar', 1)
-    assert len(info) == 0
+    assert info is None
 
 
 def test_get_calibration_polynomial(idb):
-    info = idb.get_calibration_polynomial('CIX00036TM')
-    assert len(info) == 1
-    assert len(info[0]) == 5
+    poly = idb.get_calibration_polynomial('CIX00036TM')
```
Comment (samaloney) [inline stixcore/idb/tests/test_idb.py:200]: Is the actual calibration tested e.g. `assert poly(1) == ?`

**Review by samaloney — APPROVED** [review]
Pretty slick 👍

File: stixcore/idb/idb.py [hunk 5]
```diff
@@ -138,6 +140,83 @@ def is_variable(self):
...
+            self.A = [float(row) for row in rows[0]]
+            self.valid = True
+            self.orig = rows
+        except (ValueError, IndexError):
+            self.valid = False
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({self.orig})'
+
+    def __call__(self, x):
```
[the remark this answers is not in the document]
Comment (nicHoch) [reply stixcore/idb/idb.py:163]: yes at the moment it is a scalar only.
i was working first on a hybrid solution. I will add generic array support if needed.

File: stixcore/idb/manager.py (hunk 3 above)
Comment (nicHoch) [reply stixcore/idb/manager.py:250]: this part is not really flexible at the moment will wait for  https://github.com/i4Ds/STIXCore/issues/59 to clean it up

**Pull request merged**